### ISSUE TITLE
requestAnimationFrame instead of setIinterval

### ISSUE THREAD

**Issue by oankarberg** [chafey/cornerstoneTools#186]
Hi,

Shouldn't requestAnimationFrame be used instead of setInterval when playing a clip of stacks? setInterval doesn't consider the rendering pipeline of the browser and will therefor affect the performance. The browser will automatically take care of when and how its best to run requestAnimationFrame according to the browsers rendering pipeline. 
I ran the same CT stacks with requestAnimationFrame and setInterval. SetInterval showed some "Long Frame" (about 30 fps instead of 60) during garbage collection. 

requestAnimationFrame statistics, continously 60 fps: 
https://gyazo.com/6c24f5c9d9cd25cecbc855f4f93894b2

setInterval, bumps of 30 fps close to garbage collection: 
https://gyazo.com/c5a13b0f857ec538b0f8550d2188c03c

**Comment by jpambrun** [comment]
Yes, using requestAnimationFrame across the board is planned for the near term.

**Comment by swederik** [comment]
Does this seem like a suitable approach? https://github.com/chafey/cornerstoneTools/pull/195

I used some drop-in replacements for setTimeout and setInterval that use requestAnimationFrame (https://gist.github.com/joelambert/1002116).

**Comment by dannyrb** [comment]
It looks like this is still in the works.

https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/stackTools/playClip.js#L245-L255

Any news, @swederik?
